stage/stage_utils.go

```diff
@@ -234,11 +245,13 @@ func (s *Stage) propagateStates() {
 				nextStage.SessionParams[k] = v
 			}
 		}
-		if nextStage.ColdRuns == 0 {
+		if nextStage.ColdRuns == nil && nextStage.WarmRuns == nil {
 			nextStage.ColdRuns = s.ColdRuns
-		}
-		if nextStage.WarmRuns == 0 {
 			nextStage.WarmRuns = s.WarmRuns
+		} else if nextStage.ColdRuns == nil {
+			nextStage.ColdRuns = &DefaultColdRunsValue
```

Comment:
```suggestion
			nextStage.ColdRuns = &RunsValueZero
```

If `nextStage` has only set warm run = 1 but left cold run uninitialized, the intention is to leave cold run = 0 instead of "default cold value" which is 1.

This is why I feel it is better we rename those variables from `DefaultColdRunsValue` to `RunsValueOne`. Because in some scenarios you do not want to set the default value for cold run to 1.

Hope that makes sense to you.

Reply:
Made changes as the discussion:
1. Set the default value to 0 for both of cold/warm run if it's not defined.
2. Update the constant names
3.  If both of cold/warm run with 0,  then cold run is set to 1.

Reply:
@ethanyzhang ,  I added a json validator for cold_runs/warm_runs, only integer >=0 is allowed. With invalid values, it will popup errors in json parse stage. Please help review the code changes.
Also, the validator is a common validator. We are able to validate other fields in the future. 
The configurations are included in multiple json files probably,  no parameter is mandatory in a single json file. So there is no "required" for any parameter.